Lib/ast.py

```diff
@@ -449,8 +451,14 @@ def _compare_fields(a, b):
         if a._fields != b._fields:
             return False
         for field in a._fields:
-            a_field = getattr(a, field)
-            b_field = getattr(b, field)
+            a_field = getattr(a, field, sentinel)
+            b_field = getattr(b, field, sentinel)
+            if a_field is sentinel and b_field is sentinel:
+                # both nodes are missing a field at runtime
+                return True
+            if a_field is sentinel or b_field is sentinel:
+                # one of the node is missing a field
+                return False
```

Comment:
Why is this needed? Is it related to `__replace__`?

Reply:
The part on the 'attributes' is indeed needed but actually, it's not a `return True` that I need but a `continue` here.

The reason why it's needed is because if I write something like:
```
import ast
ast.compare(ast.Tuple(), ast.Tuple(), compare_attributes=1)
```

I get

```sh
Traceback (most recent call last):
  File "<python-input-3>", line 1, in <module>
    ast.compare(ast.Tuple(), ast.Tuple(), compare_attributes=1)
    ~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/lib/python/cpython/Lib/ast.py", line 486, in compare
    if compare_attributes and not _compare_attributes(a, b):
                                  ~~~~~~~~~~~~~~~~~~~^^^^^^
  File "/lib/python/cpython/Lib/ast.py", line 472, in _compare_attributes
    a_attr = getattr(a, attr)
AttributeError: 'Tuple' object has no attribute 'lineno'
``` 

And similarly, the same error would be there if I remove a field at runtime. I can cherry pick the fix (and the fix of the fix) and make a separate PR if you prefer though.

Reply:
I think so IMO. I'll make a separate issue for that then! (I don't think it should be backported since `ast.compare` was introduced in 3.14 (I think?). If the feature made it to 3.13, then it should be backported)

Reply:
See https://github.com/python/cpython/issues/121210 (PR: https://github.com/python/cpython/pull/121211).